commons/ava_testsuite/rpc_manager.go

```diff
@@ -0,0 +1,147 @@
+	client := rpcManager.client
+	username := rpcManager.rpcUser.username
+	password := rpcManager.rpcUser.password
+	_, err := client.KeystoreApi().CreateUser(username, password)
+	if err != nil {
+		stacktrace.Propagate(err, "Could not create user.")
+	}
+	_, err = client.KeystoreApi().CreateUser(GENESIS_USERNAME, GENESIS_PASSWORD)
+	if err != nil {
+		stacktrace.Propagate(err, "Could not create genesis user.")
+	}
+	nodeId, err := client.AdminApi().GetNodeId()
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Could not get node id")
+	}
+	genesisAccountAddress, err := client.XChainApi().ImportKey(
+		GENESIS_USERNAME,
+		GENESIS_PASSWORD,
+		rpcManager.testNet.FundedAddresses.PrivateKey)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Failed to take control of genesis account.")
+	}
+	logrus.Debugf("Adding Node %s as a validator.", nodeId)
+	logrus.Debugf("Genesis Address: %s.", genesisAccountAddress)
+	testAccountAddress, err := client.XChainApi().CreateAddress(username, password)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Failed to create address on XChain.")
+	}
+	logrus.Debugf("Test account address: %s", testAccountAddress)
+	txnId, err := client.XChainApi().Send(amount, "AVA", testAccountAddress, GENESIS_USERNAME, GENESIS_PASSWORD)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Failed to send AVA to test account address %s", testAccountAddress)
+	}
+	err = rpcManager.waitForTransactionAcceptance(txnId)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Failed to wait for transaction acceptance.")
+	}
+	return testAccountAddress, nil
+}
+
+/*
+	Creates a new account on the PChain under the username and password.
+	Transfers funds from an XChain account owned by that username and password to the new PChain account.
+	Returns the new, funded PChain account address.
+*/
+func (rpcManager RpcManager) TransferAvaXChainToPChain(
+		amount int) (string, error) {
+	client := rpcManager.client
+	username := rpcManager.rpcUser.username
+	password := rpcManager.rpcUser.password
+	pchainAddress, err := client.PChainApi().CreateAccount(username, password, nil)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Failed to create new account on PChain")
+	}
+	txnId, err := client.XChainApi().ExportAVA(pchainAddress, amount, username, password)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Failed to export AVA to pchainAddress %s", pchainAddress)
+	}
+	err = rpcManager.waitForTransactionAcceptance(txnId)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "")
+	}
+	txnId, err = client.PChainApi().ImportAVA(username, password, pchainAddress, rpcManager.rpcUser.incrementNonce())
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Failed import AVA to pchainAddress %s", pchainAddress)
+	}
+	_, err = client.PChainApi().IssueTx(txnId)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "Failed to issue importAVA transaction.")
+	}
+	time.Sleep(time.Second * 5)
```

Comment:
Doesn't seem like we should be sleeping here - instead using a more precise method